packages/components-native/src/Divider/Divider.tsx

```diff
@@ -1,22 +1,40 @@
 import React from "react";
 import { View } from "react-native";
-import { styles } from "./Divider.style";
+import { horizontalStyles } from "./DividerHorizontal.style";
+import { verticalStyles } from "./DividerVertical.style";
 
 interface DividerProps {
   /**
    * The weight of the divider.
    *
    * @default "base"
    */
-  readonly size?: keyof typeof styles;
+  readonly size?: "base" | "large" | "larger" | "largest";
+  /**
+   * The direction of the divider
+   *
+   * @default "horizontal"
+   */
+  readonly direction?: "horizontal" | "vertical";
+  /**
+   * Used to locate this view in end-to-end tests.
+   */
+  readonly testID?: string;
 }
 
-export function Divider({ size = "base" }: DividerProps): JSX.Element {
+export function Divider({
+  size = "base",
+  direction = "horizontal",
+  testID = "Divider",
+}: DividerProps): JSX.Element {
+  const directionalStyles =
+    direction === "horizontal" ? horizontalStyles : verticalStyles;
   const style = [
-    styles.base,
-    size === "large" && styles.large,
-    size === "largest" && styles.largest,
+    directionalStyles.base,
```

Comment:
This would work okay for vertical, but for the horizontal I would need to set the right borders to 0 and apply styling to the bottom border instead. I did originally have a base style and set both borders, then removed the other directional border but thought that was confusing.

Reply:
Ah my bad, I misread the styles. I agree that it is better left the way it is now